### PR TITLE
Remove custom random utilities

### PR DESCRIPTION
**Pull request by adriansmares**
#### Summary

This PR removes some custom random utilities that we've had in our code base over the years.

#### Changes

- Remove the `randutil` package
- Remove some of the remaining populators
- Remove usages of `math/random` outside of tests

#### Testing

Unit testing.

##### Regressions

N/A

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/identityserver/store/populate.go [hunk 1]
```diff
@@ -25,14 +25,13 @@ import (
...
 	"go.thethings.network/lorawan-stack/v3/pkg/types"
-	"go.thethings.network/lorawan-stack/v3/pkg/util/randutil"
 )
 
 // NewPopulator returns a new database populator with a population of the given size.
 // It is seeded by the given seed.
 func NewPopulator(size int, seed int64) *Populator {
 	now := time.Now()
-	randy := randutil.NewLockedRand(rand.NewSource(seed))
+	randy := rand.New(rand.NewSource(seed))
```
Comment (adriansmares) [inline pkg/identityserver/store/populate.go:34]: This is used only locally, and not concurrently - no point in locking.

File: pkg/crypto/encryption_test.go [hunk 2]
```diff
@@ -45,9 +45,8 @@ func TestEncryptDecrypt(t *testing.T) {
 	} {
 		t.Run(fmt.Sprintf("%s", tc.Name), func(t *testing.T) {
 			a := assertions.New(t)
-			rand := random.New()
 			var key types.AES128Key
-			err := key.UnmarshalBinary(rand.Bytes(16))
+			err := key.UnmarshalBinary(random.Bytes(16))
```
Comment (htdvisser) [inline pkg/crypto/encryption_test.go]: Shouldn't we make this consistent with the other files?

```suggestion
			rand.Read(key[:])
```

**Review by htdvisser — APPROVED** [review]
LGTM

**Review by kschiffer — APPROVED** [review]
Codeowned LGTM

**Review by johanstokking — APPROVED** [review]
Code owned files LGTM

**Pull request merged**